File: src/main/java/pl/training/workshop/m6/s09_observer/step2/PaymentServices.java
package pl.training.workshop.m6.s09_observer.step2;

import pl.training.workshop.m6.s09_observer.LoyaltyProgram;
import pl.training.workshop.m6.s09_observer.Mailer;
import pl.training.workshop.m6.s09_observer.SmsGateway;

/** Krok 2: bez zmian - korzeń kompozycji. */
public final class PaymentServices {
    private PaymentServices() {
    }

    public static PaymentService standard(Mailer mailer, SmsGateway sms, LoyaltyProgram loyalty) {
        return new PaymentService(mailer, sms, loyalty);
    }
}

File: typescript/src/workshop/m6/s09_observer/step1/PaymentServices.ts
import type { LoyaltyProgram } from '../LoyaltyProgram.js';
import type { Mailer } from '../Mailer.js';
import type { SmsGateway } from '../SmsGateway.js';
import { PaymentService } from './PaymentService.js';

/** Krok 1: bez zmian - korzeń kompozycji. */
export class PaymentServices {
  private constructor() {}

  static standard(mailer: Mailer, sms: SmsGateway, loyalty: LoyaltyProgram): PaymentService {
    return new PaymentService(mailer, sms, loyalty);
  }
}
